cci_nepal/analysis/Data Analysis of Hill Data.py

```diff
@@ -0,0 +1,653 @@
+
+# #### Diving into extreme observations ####
+
+# Here, we will look into observations that have found even Male Dhoti and Whistle Blow, two items deemed least essential overall as essential.
+
+# #### Male Dhoti i.e Lower #####
+
+# In[49]:
+
+
+lower_df = df_clean.loc[df_clean["Male.Dhoti"] == 3]
+
+
+# In[50]:
+
+
+lower_df["What.is.the.Ethnicity.of.your.household."].value_counts(normalize=True)
+
+
+# In[51]:
+
+
+df_clean["What.is.the.Ethnicity.of.your.household."].value_counts(normalize=True)
+
+
+# Compared to 8 percent representation of total respondents number, Madhesi ethnicity has 25 percent representation of respondents deeming Male Dhoti as highly essential. So, a good remark in terms of validating the quality of survey response.
+
+# In[52]:
+
+
+lower_preferences_basic = (
+    lower_df.loc[:, nlf.nfri_basic].apply(pd.Series.value_counts, normalize=True)
+).T
+lower_preferences_basic.sort_values(by=[3], inplace=True, ascending=False)
+lower_preferences_basic.rename(
+    columns={1: "Unnecessary", 2: "Desirable", 3: "Essential"}, inplace=True
+)
+lower_preferences_basic
+
+
+# In[53]:
+
+
+lower_preferences_non_basic = (
+    lower_df.loc[:, nlf.nfri_non_basic].apply(pd.Series.value_counts, normalize=True)
+).T
+lower_preferences_non_basic.sort_values(by=[3], inplace=True, ascending=False)
+lower_preferences_non_basic.rename(
+    columns={1: "Unnecessary", 2: "Desirable", 3: "Essential"}, inplace=True
+)
+lower_preferences_non_basic
+
+
+# Understandably, those who deem even Male Dhoti essential deem most of other items as essential too.
+
+# #### Whistle Blow ####
+
+# In[54]:
+
+
+whistle_df = df_clean.loc[df_clean["Whistle.Blow"] == 3]
+
+
+# In[55]:
+
+
+whistle_df["What.is.the.Ethnicity.of.your.household."].value_counts(normalize=True)
+
+
+# In[56]:
+
+
+whistle_df["What.is.the.Gender.of.the.respondent."].value_counts(normalize=True)
+
+
+# Interestingly, here also the Madhesi ethnicity is hugely represented (more than 4 times in total data). It will be interesting to compare this preference when we receive Mahottari data.
+
+# In[57]:
+
+
+whistle_preferences_basic = (
+    whistle_df.loc[:, nlf.nfri_basic].apply(pd.Series.value_counts, normalize=True)
+).T
+whistle_preferences_basic.sort_values(by=[3], inplace=True, ascending=False)
+whistle_preferences_basic.rename(
+    columns={1: "Unnecessary", 2: "Desirable", 3: "Essential"}, inplace=True
+)
+whistle_preferences_basic
+
+
+# In[58]:
+
+
+whistle_preferences_non_basic = (
+    whistle_df.loc[:, nlf.nfri_non_basic].apply(pd.Series.value_counts, normalize=True)
+).T
+whistle_preferences_non_basic.sort_values(by=[3], inplace=True, ascending=False)
+whistle_preferences_non_basic.rename(
+    columns={1: "Unnecessary", 2: "Desirable", 3: "Essential"}, inplace=True
+)
+whistle_preferences_non_basic
+
+
+# As above, those who deem even Whistle Blow essential deem most of other items as essential too.
```

Comment:
Looks good and you've covered off alot of exploratory steps. Might be worth documenting the main findings in mark down at the bottom as bullet pointed list as there is alot of there so it would be helpful to read that at the bottom.